impeller/core/host_buffer.h

```diff
@@ -6,20 +6,29 @@
 #define FLUTTER_IMPELLER_CORE_HOST_BUFFER_H_
 
 #include <algorithm>
+#include <array>
+#include <functional>
 #include <memory>
 #include <string>
 #include <type_traits>
 
-#include "impeller/base/allocation.h"
 #include "impeller/core/buffer.h"
 #include "impeller/core/buffer_view.h"
 #include "impeller/core/platform.h"
 
 namespace impeller {
 
-class HostBuffer final : public Buffer {
+/// Approximately the same size as the max frames in flight.
+static const constexpr size_t kHostBufferArenaSize = 3u;
```

Comment:
No need to do this now -- I think this is fine and I'm sure it'll work, but longterm we should probably have a way to query the max frames in flight for the backend... Even though this is technically tied to the surface and not really intrinsic to the whole backend. Something like `impeller::Context::GetMaxFramesInFlight()`...

I'm also guilty of using 3 as a constant in Flutter GPU utilities (currently present in the example repos, not checked in as part of the API yet).